CONTRIBUTING.md

```diff
@@ -0,0 +1,133 @@
+# Contributing to opentelemetry-go
+
+The Go special interest group (SIG) meets regularly. See the
+OpenTelemetry
+[community](https://github.com/open-telemetry/community#golang-sdk)
+repo for information on this and other language SIGs.
+
+See the [public meeting
+notes](https://docs.google.com/document/d/1A63zSWX0x2CyCK_LoNhmQC4rqhLpYXJzXbEPDUQ2n6w/edit#heading=h.9tngw7jdwd6b)
+for a summary description of past meetings. To request edit access,
+join the meeting or get in touch on
+[Gitter](https://gitter.im/open-telemetry/opentelemetry-go).
+
+## Development
+
+There are some generated files checked into the repo. To make sure
+that the generated files are up-to-date, run `make` (or `make
+precommit` - the `precommit` target is the default).
+
+The `precommit` target also fixes the formatting of the code and
+checks the status of the go module files.
+
+If after running `make precommit` the output of `git status` contains
+`nothing to commit, working tree clean` then it means that everything
+is up-to-date and properly formatted.
+
+## Pull Requests
+
+### How to Send Pull Requests
+
+Everyone is welcome to contribute code to `opentelemetry-go` via
+GitHub pull requests (PRs).
+
+To create a new PR, fork the project in GitHub and clone the upstream
+repo:
+
+```sh
+$ go get github.com/open-telemetry/opentelemetry-go
+```
+
+(This may print some warning about "build constraints exclude all Go
+files", just ignore it.)
+
+This will put the project in
+`${GOPATH}/src/github.com/open-telemetry/opentelemetry-go`. You can
+alternatively use `git` directly with:
+
+```sh
+$ git clone https://github.com/open-telemetry/opentelemetry-go
+```
+
+This would put the project in the `opentelemetry-go` directory in
+current working directory.
+
+Enter the newly created directory and add your fork as a new remote:
+
+```sh
+$ git remote add <YOUR_FORK> git@github.com:<YOUR_GITHUB_USERNAME>/opentelemetry-go
+```
+
+Check out a new branch, make modifications, run linters and tests, and
+push the branch to your fork:
+
+```sh
+$ git checkout -b <YOUR_BRANCH_NAME>
+# edit files
+$ make precommit
```

Comment:
precommit will fail without necessary tools (golangci-lint, misspell, etc)
May be add install-tools target in Makefile.

Reply:
OOooh. An install-tools target would be great. I definitely ran into this when trying to get started the first time.

Reply:
That rather sounds like a bug - I'd like to see an output from `make precommit` with a failure about the missing tools, because it should normally build the tools if they are missing. The `precommit` target depends on the every tool…

```
$ git clean -ffdx
$ make precommit
go build -o ./.tools/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
go build -o ./.tools/misspell github.com/client9/misspell/cmd/misspell
go build -o ./.tools/stringer golang.org/x/tools/cmd/stringer
PATH="/home/kv/projects/lightstep/opentelemetry-go/ci-unclean-msg/.tools:${PATH}" go generate ./...
```

So I'd prefer avoiding adding this.

Reply:
```
➜  export GOPATH=/Users/rahulpa/rghetia/go/new
➜  go get go.opentelemetry.io
package go.opentelemetry.io: build constraints exclude all Go files in /Users/rahulpa/rghetia/go/new/src/go.opentelemetry.io
➜  cd $GOPATH/src/go.opentelemetry.io
➜  go.opentelemetry.io git:(master) make precommit
go build -o ./.tools/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
can't load package: package github.com/golangci/golangci-lint/cmd/golangci-lint: cannot find package "github.com/golangci/golangci-lint/cmd/golangci-lint" in any of:
	/usr/local/Cellar/go/1.12.4/libexec/src/github.com/golangci/golangci-lint/cmd/golangci-lint (from $GOROOT)
	/Users/rahulpa/rghetia/go/new/src/github.com/golangci/golangci-lint/cmd/golangci-lint (from $GOPATH)
make: *** [.tools/golangci-lint] Error 1
➜  go.opentelemetry.io git:(master)
```

Reply:
Thanks. I can reproduce it with go 1.12, but not with go 1.13. Let's see if I can get some workaround.

Reply:
Updated the PR again. Basically go older than 1.13 + repo in GOPATH = ignore modules. I added the commit to force the modules to be used. This should make things work.